### PR TITLE
Adding modules and updating JavaFX version to LTS.

### PR DESCRIPTION
**Pull request by mbcoder**
Adding modules and updating JavaFX version to LTS.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -41,8 +41,13 @@ dependencies {
 }
 
 javafx {
-    version = "11.0.1"
-    modules = [ 'javafx.controls' ]
+    version = "11.0.2"
```
Comment (colinanderson) [inline build.gradle:44]: Shouldn't it be 11.0.6 to avoid that HTTP problem?

Comment (mbcoder) [reply build.gradle:44]: @colinanderson this is the JavaFX version, not the JDK/JRE version

11.0.2 is LTS version of JavaFX

Comment (mbcoder) [reply build.gradle:44]: @colinanderson actually I'm wrong, not LTS, but public version is 11.0.2.

https://gluonhq.com/products/javafx/

File: build.gradle [hunk 2]
```diff
@@ -60,5 +65,5 @@ run {
 }
 
 wrapper {
-    gradleVersion = '5.0'
+    gradleVersion = '6.5.1'
 }
```
Comment (mbcoder) [inline build.gradle:69]: @colinanderson I do wonder if we need to specify the gradle version in the script when we already have the wrapper in the repository.  I guess its useful for information though...

File: build.gradle [hunk 3]
```diff
@@ -41,8 +41,13 @@ dependencies {
...
 
 javafx {
-    version = "11.0.1"
-    modules = [ 'javafx.controls' ]
+    version = "11.0.2"
+    modules = [ 'javafx.controls',
```
Comment (Rachael-E) [inline build.gradle:45]: we don't need all 6 of these as far as I am aware: across all of our samples we only use:
'javafx.controls',
'javafx.fxml',
'javafx.web',
'javafx.media'

I believe .base and .graphics are used internally though but not needed for building an application.

File: gradlew [hunk 4]
```diff
@@ -1,18 +1,20 @@
 #!/usr/bin/env sh
 
-#Copyright 2019 Esri
 #
-#Licensed under the Apache License, Version 2.0 (the "License"); you may not
-#use this file except in compliance with the License. You may obtain a copy
-#of the License at
+# Copyright 2015 the original author or authors.
```
Comment (Rachael-E) [inline gradlew:4]: i guess this is generated everytime it's run but I wonder why it changed to 2015!

File: build.gradle (hunk 3 above)
Comment (colinanderson) [reply build.gradle:45]: I believe base and graphics are included by using something like controls so don't necessarily need to be explicitly added.

File: build.gradle (hunk 2 above)
Comment (colinanderson) [reply build.gradle:69]: I think this is only for creating the wrapper so shouldn't be needed if the wrapper files are already there.

File: build.gradle [hunk 5]
```diff
@@ -41,8 +41,13 @@ dependencies {
...
+                'javafx.base',
```
Comment (colinanderson) [inline build.gradle:46]: In a proper app I think you would only include the modules that are actually required. Looking at the app web, media and probably fxml are not required. Maybe a better approach would be to explain in the doc that you might need to add more modules depending on the functionality you use and also list all the modules that we know are used if you use every part of the API.

I think you can run jdeps or something on the JAR (might also work on a class) to get the full list. I can't remember the error message you get if you are missing a module but I think there is enough information in it to work out how to resolve it.

You can't win though - if you include them all someone will point out they aren't all needed and if you don't others will not know what to do when they add more API use and it doesn't compile.

File: build.gradle (hunk 2 above)
Comment (colinanderson) [reply build.gradle:69]: I think the wrapper task is only needed if you want to create a wrapper and isn't needed if you already have the wrapper.

**Pull request closed without merging**